src/otx/core/model/module/base.py

```diff
@@ -90,11 +90,13 @@ def training_step(self, inputs: OTXBatchDataEntity, batch_idx: int) -> Tensor:
     def on_validation_epoch_start(self) -> None:
         """Callback triggered when the validation epoch starts."""
         if self.metric:
+            self.metric.num_classes = self.model.num_classes
```

Comment:
@sungmanc, please change this in your PR such as
```suggestion
            num_classes_augmented_params = {
                 param.default if name != "num_classes" else self.model.num_classes
                 for name, param in inspect.signature(metric).parameters.items()
                 if not name != "kwargs"
            }
            metric = metric(**num_classes_augmented_params)
```